src/broker/WebSocketShim.cc

```diff
@@ -180,3 +180,247 @@ void WebSocketShim::ProcessMessage(std::string_view topic, broker::zeek::Invalid
 }
 
 } // namespace zeek::Broker
+
+
+/*
+ * The below tests are sanity tests for broker's endpoints, hubs and subscriber functionality
+ * required by the WebSocketShim implementation.
+ */
+#include "broker/message.hh"
+#include "broker/publisher.hh"
+
+#include "zeek/3rdparty/doctest.h"
+
+namespace {
+
+TEST_SUITE_BEGIN("broker-websocket-shim");
+using namespace std::literals;
+
+TEST_CASE("tests") {
+    // broker::set_console_logger("debug");
+    broker::broker_options opts;
+    opts.skip_ssl_init = true;
+    opts.disable_forwarding = true;
+    opts.disable_ssl = true;
+    broker::configuration conf{opts};
+    auto ep = broker::endpoint(std::move(conf));
+
+    auto hub1 = ep.make_hub({"/abc"});
+    auto hub2 = ep.make_hub({"/abc", "/cde"});
+    auto sub1 = ep.make_subscriber({"/abc"});
+    auto sub2 = ep.make_subscriber({"/abc", "/cde"});
+
+    auto pub1 = ep.make_publisher(broker::topic{"/abc"});
+    auto pub2 = ep.make_publisher(broker::topic{"/abc"});
+
+    auto make_message = [](int value) { return broker::list_builder{}.add(value).build(); };
+
+    auto make_data_message = [](auto topic, int value) {
+        auto msg = broker::list_builder{}.add(value).build();
+        return broker::make_data_message(broker::topic{topic}, msg);
+    };
+
+    SUBCASE("endpoint-publish") {
+        // Publishing through the endpoint is visible to hubs, but not subscribers.
```

Comment:
Thanks for adding this here, very helpful. I wish we had this over in the Broker docs, where I spent a whole bunch of time earlier today trying to figure out [exactly this](https://github.com/zeek/broker/pull/451#discussion_r2055112675).